internal/utils.go

```diff
@@ -0,0 +1,35 @@
+package internal
+
+import (
+	"github.com/hamba/avro/v2"
+)
+
+func MustNewUnionSchema(types []avro.Schema, opts ...avro.SchemaOption) *avro.UnionSchema {
+	s, err := avro.NewUnionSchema(types, opts...)
+
+	if err != nil {
+		panic(err)
+	}
+
+	return s
+}
```

Comment:
these can all be simplified with a simple

```go
func Must[T any](v T, err error) T {
    if err != nil {
        panic(err)
    }
    return v
}
```

and then just using that generic method around the rest of the calls instead. Honestly, we probably can just have that utility function and not bother having these at all, just calling `Must(avro.NewUnionSchema(.....))` etc.